core/test/integ/src/plugins/kubernetes/container/build/build.ts

```diff
@@ -566,3 +579,168 @@ describe.skip("Kubernetes Container Build Extension", () => {
     })
   })
 })
+
+describe("Ensure serviceAccount annotations for in-cluster building", () => {
+  let garden: Garden
+  let cleanup: (() => void) | undefined
+  let log: ActionLog
+  let provider: KubernetesProvider
+  let ctx: PluginContext
+  let api: KubeApi
+  after(async () => {
+    if (garden) {
+      garden.close()
+    }
+  })
+
+  const init = async (environmentName: string, remoteContainerAuth = false) => {
+    ;({ garden, cleanup } = await getContainerTestGarden(environmentName, { remoteContainerAuth }))
+    log = createActionLog({ log: garden.log, actionName: "", actionKind: "" })
+    provider = <KubernetesProvider>await garden.resolveProvider(garden.log, "local-kubernetes")
+    ctx = await garden.getPluginContext({ provider, templateContext: undefined, events: undefined })
+    api = await KubeApi.factory(log, ctx, provider)
+  }
+  grouped("kaniko").context("kaniko service account annotations", () => {
+    beforeEach(async () => {
+      await init("kaniko")
+    })
+    it("should deploy a garden builder serviceAccount with specified annotations in the project namespace", async () => {
+      const annotations = {
+        "iam.gke.io/gcp-service-account": "workload-identity-gar@garden-ci.iam.gserviceaccount.com",
+      }
+      const projectNamespace = ctx.namespace
+      provider.config.kaniko = { serviceAccountAnnotations: annotations }
+      const serviceAccount = getBuilderServiceAccountSpec(projectNamespace, annotations)
+
+      await ensureServiceAccount({ ctx, log, api, namespace: projectNamespace, annotations })
+
+      const status = await compareDeployedResources({
+        ctx: ctx as KubernetesPluginContext,
+        api,
+        namespace: projectNamespace,
+        manifests: [serviceAccount],
+        log,
+      })
+      expect(status.state).to.equal("ready")
+    })
+    it("should remove annotations from the garden builder serviceAccount", async () => {
+      const projectNamespace = ctx.namespace
+      const originalAnnotations = {
+        "iam.gke.io/gcp-service-account": "workload-identity-gar@garden-ci.iam.gserviceaccount.com",
+        "foo": "bar",
+      }
+      provider.config.kaniko = { serviceAccountAnnotations: originalAnnotations }
+      const originalServiceAccount = getBuilderServiceAccountSpec(projectNamespace, originalAnnotations)
+
+      await ensureServiceAccount({ ctx, log, api, namespace: projectNamespace, annotations: originalAnnotations })
+
+      const status = await compareDeployedResources({
+        ctx: ctx as KubernetesPluginContext,
+        api,
+        namespace: projectNamespace,
+        manifests: [originalServiceAccount],
+        log,
+      })
+      // Both annotations should be present
+      expect(isEqualAnnotations(originalServiceAccount, status.remoteResources[0])).to.be.true
```

Comment:
I'm ok with this, but if the test fails this will give a better error message with the actual difference:
```suggestion
      expect(isEqualAnnotations(originalServiceAccount)).to.equal(status.remoteResources[0])
```

Reply:
This doesn't really work like that since `isEqualAnnotations` is a function that takes two parameters, and it checks the following
```
  const a1 = r1.metadata.annotations !== undefined ? r1.metadata.annotations : {}
  const a2 = r2.metadata.annotations !== undefined ? r2.metadata.annotations : {}
  return isEqual(a1, a2)
```

So I think what would be needed is 
```
expect(originalServiceAccount.metadata.annotations).to.eql(status.removeResources[0].metadata.annotations)
```

`to.equal` is strict equality btw so not a good match here.

Reply:
I see that change was already fixed so I'll go ahead and resolve that now :)

Reply:
Ah yes, my snippet was flawed but anna correctly used deep.equal on just the annotations; That should work fine 👍